[WPE] Upgrade our custom run-benchmark plans to use MotionMark 1.3.1

nikolaszimmermann opened the pull request:
#### de6e87fb4c3cce308f3326c57a2cb528241c454f
<pre>
[WPE] Upgrade our custom run-benchmark plans to use MotionMark 1.3.1
<a href="https://bugs.webkit.org/show_bug.cgi?id=280051">https://bugs.webkit.org/show_bug.cgi?id=280051</a>

Reviewed by Simon Fraser and Carlos Garcia Campos.

Switch our custom run-benchmark plans to use MotionMark 1.3.1.
This has the main benefit that we can run individual subtests,
and get the right &apos;canvas size&apos;, which is one of the 1.3.1
fixes.

Furthermore rework the 15 FPS logic, to unbreak score computation.
The expectation is that the cadence of the rAF() callback matches the
screen refresh rate -- however we still want to run at 60 Hz refresh
rate, but artifically limit the complexity, to get higher scores on
embedded devices (e.g. a few hundred points, instead of scores between
1-4 depending on the subtest). This increases the likelihood to detect
regressions/progresses, even on low-end devices.

Realize this by injecting a rAF() proxy, which only calls the MotionMark
rAF() logic, every 4th callback, simulating 15 FPS.

* Tools/Scripts/webkitpy/benchmark_runner/data/patches/extra/MotionMark1.3.1-15FPS.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/extra/MotionMark1.3.1-60FPS.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3.1-15fps.plan: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3.1-60fps.plan: Added.

Canonical link: <a href="https://commits.webkit.org/284147@main">https://commits.webkit.org/284147@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/01dbef192a33e3551df76326d7a61081dcbb915f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/68593 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/47985 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21252 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/72662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/19738 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/70710 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/55781 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/19554 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/72662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13126 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-layout-tests-in-stress-mode; Passed layout tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/71660 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43852 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/59229 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/72662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/68104 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/40519 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/16652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18095 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/61711 "Build is in progress. Recent messages:") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62471 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/16999 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/74356 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/12564 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16249 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/62180 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/12604 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/59308 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/62205 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15201 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10145 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/3766 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/43786 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/44860 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46054 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/44602 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

clopez reviewed (changes requested):
Change looks ok, but we need to also rename the files to be consistent

So file `motionmark1.3-60fps.plan` should be renamed as `motionmark1.3.1-60fps.plan` and same for `-15fps` version

And I would also rename the patch files to contain the `1.3.1` string on it.

And on the runner we later have to modify the name of the plan from `motionmark-1.3-60fps` to `motionmark-1.3.1-60fps`  and same for `-15fps` version

This is a new version of the benchmark so it should be reflected as a change on the plan name

If we want to keep the old version around (1.3) instead of renaming the files we can add new ones. But i don't see a need to keep the old version so we can just rename them.

By the way, we also should change the plan name of the default motionmark that we pass to the runner, now it should be `motionmark-1.3.1` instead of `motionmark-1.3`

nikolaszimmermann committed:
[WPE] Upgrade our custom run-benchmark plans to use MotionMark 1.3.1
https://bugs.webkit.org/show_bug.cgi?id=280051

Reviewed by Simon Fraser and Carlos Garcia Campos.

Switch our custom run-benchmark plans to use MotionMark 1.3.1.
This has the main benefit that we can run individual subtests,
and get the right 'canvas size', which is one of the 1.3.1
fixes.

Furthermore rework the 15 FPS logic, to unbreak score computation.
The expectation is that the cadence of the rAF() callback matches the
screen refresh rate -- however we still want to run at 60 Hz refresh
rate, but artifically limit the complexity, to get higher scores on
embedded devices (e.g. a few hundred points, instead of scores between
1-4 depending on the subtest). This increases the likelihood to detect
regressions/progresses, even on low-end devices.

Realize this by injecting a rAF() proxy, which only calls the MotionMark
rAF() logic, every 4th callback, simulating 15 FPS.

* Tools/Scripts/webkitpy/benchmark_runner/data/patches/extra/MotionMark1.3.1-15FPS.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/extra/MotionMark1.3.1-60FPS.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3.1-15fps.plan: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3.1-60fps.plan: Added.

Canonical link: https://commits.webkit.org/284147@main
